src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -670,9 +673,14 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                         throw ExceptionUtilities.UnexpectedValue(_deconstruction.Kind());
                 }
 
-                TypeSymbol result = this._type;
-                Debug.Assert((object)result != null);
-                return result;
+                if ((object)this._type == null)
+                {
+                    diagnostics.Add(this.ForbiddenDiagnostic, Locations[0], Name);
```

Comment:
I don't know of a way to trigger this defensive code. I'll remove it and let the assertion in the caller fail if it occurs.

---
In reply to: [91395018](https://github.com/dotnet/roslyn/pull/15548#discussion_r91395018) [](ancestors = 91395018)